src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchResultsProcessor.kt

```diff
@@ -4,101 +4,138 @@ import com.microsoft.applicationinsights.TelemetryClient
-
     val prisoners = prisonerMatches.matches.map { it.prisoner }
-    val prisonerMatchCount = PrisonerMatchCount.of(prisoners, prisonerSearchRequest)
-    val customDimensions = PrisonerSearchCustomDimensions(
-      dataSource = MATCH_PRISONERS,
-      searchType = prisonerSearchRequest.searchType,
-      numberOfResults = prisoners.size,
-      exactMatchCount = prisonerMatchCount.mainDetailsCount,
-      aliasExactMatchCount = prisonerMatchCount.aliasDetailsCount
-    ).asMap()
-
-    telemetryClient.trackEvent("prisoner-search", customDimensions, null)
+    val prisonerMatch = PrisonerMatch.of(prisoners, prisonerSearchRequest)
+
+    with(
+      prisonerSearchSummaryCustomDimensions(
+        dataSource = MATCH_PRISONERS,
+        searchType = prisonerSearchRequest.searchType,
+        numberOfResults = prisoners.size,
+        exactMatchCount = prisonerMatch.mainDetailsCount,
+        aliasExactMatchCount = prisonerMatch.aliasDetailsCount
+      )
+    ) {
+      telemetryClient.trackEvent("prisoner-search-summary", this, null)
+    }
+
+    prisonerMatch.bestMatch?.also {
+      with(prisonerSearchBestMatchCustomDimensions(it, caseload)) {
+        telemetryClient.trackEvent("prisoner-search-best-match", this, null)
+      }
+    }
   }
 
   fun processSearchResults(
     pagePrisoner: PagePrisoner,
     prisonerSearchRequest: PrisonerSearchRequest,
     caseload: String
   ) {
-    prisonerSearchResultsLogger.logResults(pagePrisoner, prisonerSearchRequest, caseload)
-
     val prisoners = pagePrisoner.content ?: emptyList()
-    val prisonerMatchCount = PrisonerMatchCount.of(prisoners, prisonerSearchRequest)
-    val customDimensions = PrisonerSearchCustomDimensions(
-      dataSource = GLOBAL_SEARCH,
-      searchType = prisonerSearchRequest.searchType,
-      numberOfResults = prisoners.size,
-      exactMatchCount = prisonerMatchCount.mainDetailsCount,
-      aliasExactMatchCount = prisonerMatchCount.aliasDetailsCount
-    ).asMap()
-
-    telemetryClient.trackEvent("prisoner-search", customDimensions, null)
+    val prisonerMatch = PrisonerMatch.of(prisoners, prisonerSearchRequest)
+
+    with(
+      prisonerSearchSummaryCustomDimensions(
+        dataSource = GLOBAL_SEARCH,
+        searchType = prisonerSearchRequest.searchType,
+        numberOfResults = prisoners.size,
+        exactMatchCount = prisonerMatch.mainDetailsCount,
+        aliasExactMatchCount = prisonerMatch.aliasDetailsCount
+      )
+    ) {
+      telemetryClient.trackEvent("prisoner-search-summary", this, null)
+    }
+
+    prisonerMatch.bestMatch?.also {
+      with(prisonerSearchBestMatchCustomDimensions(it, caseload)) {
+        telemetryClient.trackEvent("prisoner-search-best-match", this, null)
+      }
+    }
   }
 }
 
 private val PrisonerSearchRequest.searchType: PrisonerSearchType
   get() = if (prisonNumber != null) PRISON_NUMBER else NAME_DOB
 
-data class PrisonerMatchCount(
```

Comment:
This class is no longer just about counts - it also contains the best matching prisoner if there is one